overlord/snapstate/handlers.go

```diff
@@ -995,15 +995,20 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 	return nil
 }
 
+func snapdSnapInstalled(st *state.State) bool {
+	var snapst SnapState
+	err := Get(st, "snapd", &snapst)
+	return err == nil && snapst.Active
```

Comment:
this has a misleading name now, because Installed and Active are different concept, I'm also not sure adding Active here is buying us anything but confusion, at worst we restart a bit more. Also the other places that check for snapd aren't checking for active either I think.

Reply:
indeed selectInterfaceMapper doesn't consider active or not (which I think is fine)

Reply:
Sorry for the confusion on my side, I changed it back again with a commit message explaining things (for the next git blame :)